mteb/models/model2vec_models.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from functools import partial
+from typing import Any
+
+import numpy as np
+
+from mteb.encoder_interface import PromptType
+from mteb.model_meta import ModelMeta
+
+from .wrapper import Wrapper
+
+logger = logging.getLogger(__name__)
+
+MAXINT = np.iinfo(np.intp).max
+
+
+class Model2VecWrapper(Wrapper):
+    def __init__(
+        self,
+        model_name: str,
+        **kwargs,
+    ) -> None:
+        """Wrapper for Model2Vec models.
+
+        Args:
+            model_name: The Model2Vec model to load from HuggingFace Hub.
+        """
+        try:
+            from model2vec import StaticModel
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                "To use the Model2Vec models `model2vec` is required. Please install it with `pip install mteb[model2vec]`."
+            ) from e
+
+        self.model_name = model_name
+        self.static_model = StaticModel.from_pretrained(self.model)
```

Comment:
Without this change, `mteb run -m minishlab/M2V_base_glove_subword -t STS12` crashes.
```suggestion
        self.static_model = StaticModel.from_pretrained(self.model_name)
```